metrics-monitoring/README.adoc

```diff
@@ -0,0 +1,225 @@
+
+This procedure follows the https://github.com/prometheus-operator/prometheus-operator/blob/main/Documentation/additional-scrape-config.md#additional-scrape-configuration[Additional Scrape Configuration] method defined by Prometheus for integrating third-party metrics.
+
+.Prerequisites
+* You have access to a Kafka instance that contains topics in {product}. For more information about access management in {product}, see {base-url}{access-mgmt-url}[_Managing account access in {product-long}_^].
+* You have the generated credentials for your service account that has access to the Kafka instance. To regenerate and re-copy the credentials, use the *Service Accounts* page in the {product} web console to find your service account and update the credentials. For more information about service accounts, see {base-url}{getting-started-url}#proc-creating-service-account_getting-started[_Getting started with {product-long}_^].
+* You or other users have produced and consumed messages in the Kafka topics. For example, to use Kafka scripts to produce and consume messages, see {base-url}{kafka-bin-scripts-url}[_Configuring and connecting Kafka scripts with {product-long}_^].
+* You've installed a Prometheus instance. For installation instructions, see https://prometheus.io/docs/prometheus/latest/getting_started/[Getting Started] in the Prometheus documentation.
+* You've installed the https://kubernetes.io/docs/tasks/tools/#kubectl[kubectl] Kubernetes command-line tool.
+
+.Procedure
+. In your Prometheus instance, create the following `kafka-federate.yaml` file. Replace the values with your own Kafka instance and credential information.
++
+--
+The `<kafka_instance_id>` is the ID of the Kafka instance. The `<client_id>` and `<client_secret>` are the generated credentials for your service account.
+
+.Example `kafka-federate.yaml` file
+[source,yaml,subs="+quotes"]
+----
+- job_name: "kafka-federate"
+  static_configs:
+  - targets: ["api.openshift.com"]
+  scheme: "https"
+  metrics_path: "/api/kafkas_mgmt/v1/kafkas/__<kafka_instance_id>__/metrics/federate"
+  oauth2:
+    client_id: "__<client_id>__"
+    client_secret: "__<client_secret>__"
+    token_url: "https://identity.api.openshift.com/auth/realms/rhoas/protocol/openid-connect/token"
+----
+
+Alternatively, you can add this information to your Prometheus configuration file, as shown in the following example. For more information about this method, see https://prometheus.io/docs/prometheus/latest/configuration/configuration/#configuration-file[Configuration File] in the Prometheus documentation.
```

Comment:
Might be good to separate this out from these steps? Once you do this, you don't really need to do Step 2 and 3 below.

Reply:
Yeah, I went back and forth on this. Separating it out felt like too much emphasis on the alternative route, and mentioning it at the end of the proc started to pile up the end of the proc with the info that's already there. But I definitely see the lack of clarity. So I've turned it into a note instead and clarified that doing that is an alternative that then negates the remaining steps. It also keeps the same setup file content together but for the two options.

Anyway, have a look at my new version of that and see what you think.

Reply:
makes sense, what you have there now looks good to me :+1: